test/xbgpu/test_engine.py

```diff
@@ -487,64 +582,75 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
             # Generate one extra chunk to simulate an incomplete accumulation
             # to check that dumps are aligned correctly - even if the first
             # received batch is from the middle of an accumulation.
-            batch_start_index = (
-                first_accumulation_index + 1
-            ) * heap_accumulation_threshold - HEAPS_PER_FENGINE_PER_CHUNK
-            batch_end_index = (first_accumulation_index + 1 + n_full_accumulations) * heap_accumulation_threshold
-            device_results = await self._send_data(
-                mock_recv_stream,
-                recv_stream,
-                heap_accumulation_threshold=heap_accumulation_threshold,
+            batch_start_index = (first_batch_index) * max_heap_acc_threshold - HEAPS_PER_FENGINE_PER_CHUNK
```

Comment:
Multiplying a batch index by a heap_acc_threshold doesn't make sense.

Reply:
Maybe I'm missing something, but ... why not?

Reply:
Is `max_heap_acc_threshold` not heaps per batch?
Doesn't change the overall thrust of your argument though, in that you then end up with units of heaps.

So there does seem to be at least some naming inconsistency there.

Reply:
I don't feel confident that the test is doing what it did in `main` (in terms of how it aligns the batches that get sent with the accumulations) since I can't easily work my way through the logic when I don't know what units the variables are in.

If you're confident that the test is doing what it's supposed to do then go ahead and fix it up later (but not much later).